File: src/cdq/ui/ctx_stage.clj
(ns cdq.ui.ctx-stage
  (:require [clojure.gdx.scenes.scene2d.group :as group])
  (:import (clojure.lang ILookup)
           (cdq.ui CtxStage)))

(defn create [viewport batch]
  (proxy [CtxStage ILookup] [viewport batch (atom nil)]
    (valAt [id]
      (group/find-actor-with-id (CtxStage/.getRoot this) id))))

(defn get-ctx [^CtxStage stage]
  @(.ctx ^CtxStage stage))

(defn set-ctx! [^CtxStage stage ctx]
  (reset! (.ctx ^CtxStage stage) ctx))

File: src/clojure/gdx/scenes/scene2d/CtxStage.java
package com.badlogic.gdx.scenes.scene2d;

import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.scenes.scene2d.Stage;
import com.badlogic.gdx.utils.viewport.Viewport;

public class CtxStage extends Stage  {

  public Object ctx;

	public CtxStage (Viewport viewport, Batch batch) {
		super(viewport, batch);
    this.ctx = null;
	}

}
